BrainPortal/app/controllers/data_providers_controller.rb

```diff
@@ -98,7 +98,7 @@ def create  #:nodoc:
       respond_to do |format|
         format.html { redirect_to :action => :index, :format => :html}
         format.xml  { render :xml   => @provider }
-        format.json { render :json  => @provider }
+        format.json { render :json  => @provider.for_api }
```

Comment:
Can you add the `.for_api` for the XML request too please? it's just that you made me realize it needs to be there also!